Marlin/src/inc/Warnings.cpp

```diff
@@ -488,3 +488,12 @@
 #if HOMING_Z_WITH_PROBE && IS_CARTESIAN && DISABLED(Z_SAFE_HOMING)
   #error "Z_SAFE_HOMING is recommended when homing with a probe. Enable Z_SAFE_HOMING or comment out this line to continue."
 #endif
+
+//
+// Warn users of potential endstop/DIAG pin conflicts to prevent homing issues when not using sensorless homing
+//
+#if !USE_SENSORLESS && USES_DIAG_JUMPERS
+  #warning "Motherboard DIAG jumpers must be removed when SENSORLESS_HOMING is disabled."
+#elif !USE_SENSORLESS && USES_DIAG_PINS
+  #warning "Driver DIAG pins must be removed when SENSORLESS_HOMING is disabled."
```

Comment:
Can pins be removed with needle-nose pliers or are flush-cutters better?

Reply:
> Can pins be removed with needle-nose pliers or are flush-cutters better?

A soldering iron would be best, but cutters included with your printer work fine.

Reply:
Hmm. Users might wonder why they need to physically remove pins entirely, especially if there is nothing connected to them.

Reply:
I think an explainer on TMC diag pin->endstop wiring might be a bit long for a warning, but pointing to various user manuals might work?

[Here's one from BTT](https://github.com/bigtreetech/BIGTREETECH-TMC2209-V1.2/blob/master/TMC2209%20problems%20solving.pdf).